safehttp/mux_test.go

```diff
@@ -191,18 +147,166 @@ func TestMuxServeTwoHandlers(t *testing.T) {
 
 func TestMuxHandleSameMethodTwice(t *testing.T) {
 	d := testDispatcher{}
-	mux := NewServeMux(d, "foo.com")
+	mux := safehttp.NewServeMux(d, "foo.com")
 
-	registeredHandler := HandlerFunc(func(w ResponseWriter, r *IncomingRequest) Result {
+	registeredHandler := safehttp.HandlerFunc(func(w safehttp.ResponseWriter, r *safehttp.IncomingRequest) safehttp.Result {
 		return w.Write(safehtml.HTMLEscaped("<h1>Hello World!</h1>"))
 	})
-	mux.Handle("/bar", MethodGet, registeredHandler)
+	mux.Handle("/bar", safehttp.MethodGet, registeredHandler)
 
 	defer func() {
 		if r := recover(); r == nil {
 			t.Errorf(`mux.Handle("/bar", MethodGet, registeredHandler) expected panic`)
 		}
 	}()
 
-	mux.Handle("/bar", MethodGet, registeredHandler)
+	mux.Handle("/bar", safehttp.MethodGet, registeredHandler)
+}
+
+type validTestInterceptor struct {
+	name  string
+	value string
+}
+
+func (p validTestInterceptor) Before(w safehttp.ResponseWriter, _ *safehttp.IncomingRequest) safehttp.Result {
+	if err := w.Header().Set(p.name, p.value); err != nil {
+		return w.ServerError(safehttp.StatusInternalServerError)
+	}
+	return safehttp.Result{}
+}
+
+type errorTestInterceptor struct{}
+
+func (errorTestInterceptor) Before(w safehttp.ResponseWriter, _ *safehttp.IncomingRequest) safehttp.Result {
```

Comment:
`errorInterceptor` -- generally, drop the "test" parts of the names is this is a `_test.go` file.